### PR TITLE
support for extended adjustable dpi

### LINKED ISSUE #2207: Logitech G Pro X Superlight 2 support
This issue doesn't really need too many details, but I'd like to get involved with adding support for the Superlight 2 mouse, which I've recently purchased. It's showing up like this:
Wired:
image
Lightspeed receiver:
image
Solaar is actually lying in the second screenshot, the polling rate reported by `evhz` is 1000 Hz, which isn't the full potential of the mouse but is also perfectly usable for gaming.

The mouse doesn't have many features, but 2000 Hz report rate on Linux and being able to toggle their hybrid switches would be great.

### PR DESCRIPTION
**Pull request by pfps**
Fixes #2207

### CODE REVIEW

File: lib/logitech_receiver/settings_templates.py [hunk 1]
```diff
@@ -817,7 +817,7 @@ def move_action(self, dx, dy):
 
 class MouseGesturesXY(_RawXYProcessing):
     def activate_action(self):
-        self.dpiSetting = next(filter(lambda s: s.name == "dpi", self.device.settings), None)
+        self.dpiSetting = next(filter(lambda s: s.name == "dp" or s.name == "dpi_extended", self.device.settings), None)
```
Comment (CuteSC2) [inline lib/logitech_receiver/settings_templates.py:820]: shouldn't it be "dpi" instead of "dp"

**Comment by pfps** [comment]
Yes indeed.

**Comment by pfps** [comment]
Yes indeed.  Should be fixed now

**Comment by CuteSC2** [comment]
grafik
It lives!!

**Comment by pfps** [comment]
Does your mouse have both X and Y dpi?  Solaar only gets and sets the X dpi and I'm not sure what this does for a mouse with both.

The Pro X 2 appears to have lots of possible dpi values.   Does this slow Solaar down appreciably? 

What happens if you try to change the DPI with an active profile?

**Comment by pfps** [comment]
What happens with the exception?   Solaar should just mark the setting as in an error state and proceed.

What is happening is that for some devices an active onboard profile prevents changes to the DPI.   Try turning off onboard profiles and see what happens.

**Comment by CuteSC2** [comment]
now a different error with onboard profiles disabled
[solar dpi error.log](https://github.com/pwr-Solaar/Solaar/files/14728969/solar.dpi.error.log)

**Comment by pfps** [comment]
My guess is that your mouse supports different resolutions for X and Y directions and that 0 is not allowable for the Y direction.  

There are several ways to support this.  I'll try the simplest - one that assumes that all valid X resolutions are also valid Y resolutions.

**Comment by pfps** [comment]
The current version of the PR makes some allowances for y direction DPI but assumes that it can be set the same as the x direction DPI.    I had thought that it would be a lot of work to separate them, but it may not be too difficult in the end.

But please try out this version and post the result of `solaar -ddd` from startup to an attempt to change DPI.

**Comment by pfps** [comment]
@MattHag I removed the pip-installed Solaar on my machine and now I'm experiencing strange effects with pytest.  
```
pytest --cov-report term-missing --cov=lib/logitech_receiver tests/logitech_receiver/test_hidpp20_simple.py 
```
generates a reasonable coverage report, but
```
pytest --cov-report term-missing --cov=lib/logitech_receiver tests/logitech_receiver/test_setting_templates.py 
``` 
produces the warning `/home/pfps/.local/lib/python3.12/site-packages/coverage/control.py:887: CoverageWarning: No data was collected. (no-data-collected)
  self._warn("No data was collected.", slug="no-data-collected")` and the coverage report shows no coverage at all even though a lot of tests were run.

Any ideas on what is going on?

**Comment by MattHag** [comment]
No, I don't know.

**Comment by DexterHaxxor** [comment]
Getting the same error as described by CuteSC2 even after these commits

**Comment by pfps** [comment]
See the bug issue for this change.

**Comment by MattHag** [comment]
Regarding the coverage that, could be caused by a segvault, which I get at least locally from one test.

**Pull request merged**